### PR TITLE
feat(*): use heredoc for adding CLI documentation

### PR DESCRIPTION
**Pull request by burntcarrot**
Closes: #121

Adds `heredoc` for CLI documentation.

Signed-off-by: burntcarrot <aadhav.n1@gmail.com>

### CODE REVIEW

File: command/report/report.go [hunk 1]
```diff
@@ -25,9 +27,22 @@ type ReportOptions struct {
...
+	y := color.New(color.FgYellow, color.Bold)
+	doc := heredoc.Docf(`
+		Report artifacts to DeepSource.
+
+		Use %[1]s to specify the analyzer, for example:
+		%[2]s
+
+		Use %[3]s to specify the value of the artifact:
+		%[4]s
+		`, y.Sprintf("--analyzer"), c.Sprintf("deepsource report --analyzer python"), y.Sprintf("--value"), c.Sprintf("deepsource report --key value"))
```
Comment (burntcarrot) [inline command/report/report.go:40]: Need more examples for this. Any suggestions?

**Comment by siddhant-deepsource** [comment]
@burntcarrot Can you please add some screenshots of how the help docs look as well?

File: command/report/report.go (hunk 1 above)
Comment (siddhant-deepsource) [reply command/report/report.go:40]: `deepsource report --analyzer go --value-file coverage.out`

`deepsource report --analyzer javascript --value-file coverage/cobertura-coverage.xml`

**Comment by burntcarrot** [comment]
> @burntcarrot Can you please add some screenshots of how the help docs look as well?

image

**Comment by siddhant-deepsource** [comment]
@burntcarrot Instead of using `color` everywhere, what do you think about creating a utility for it like [this](https://github.com/cli/cli/blob/97ea5e77b4d61d5d80ed08f7512847dee3ec9af5/pkg/iostreams/color.go)? I guess creating an abstraction/API over `color` will easen its usage across the CLI.

**Comment by burntcarrot** [comment]
@siddhant-deepsource Agreed. Moved it to `utils`.

**Review by siddhant-deepsource — APPROVED** [review]

**Review by sourya-deepsource — APPROVED** [review]

**Pull request merged**